src/vsphere_cpi/spec/integration/host_maintenance_mode_spec.rb

```diff
@@ -0,0 +1,192 @@
+                  'clusters' => [
+                      {
+                          @cluster_name_maintenance => {}
+                      },
+                  ]
+              }
+          ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    # We turn the maintenance mode ON for half of the hosts (rounding off to the floor) in a cluster
+    before do
+      turn_maintenance_on_for_half_hosts(cpi, @cluster_name_maintenance)
+    end
+    it 'cpi should be able to replicate stemcell (create vm and create ephemeral disk) on a datastore accessible by non-maintenance mode host and which belongs to same cluster' do
+      begin
+        @vm_id = cpi.create_vm(
+          'agent-007',
+          @stemcell_id,
+          vm_type,
+          get_network_spec,
+          [],
+          {}
+        )
+        expect(@vm_id).to_not be_nil
+        expect(cpi.has_vm?(@vm_id)).to be(true)
+        # These checks are sufficient as any try to create a vm on maintenance mode host or non-accessible host will error out.
+      ensure
+        delete_vm(cpi, @vm_id)
+      end
+    end
+    it 'cpi should be able to create persistent disk on a datastore accessible by non-maintenance mode host' do
+      begin
+        @disk_id = cpi.create_disk(2048, {}, nil)
+        expect(@disk_id).to_not be_nil
+        expect(cpi.has_disk?(@disk_id)).to be(true)
+        disk = cpi.datacenter.find_disk(VSphereCloud::DirectorDiskCID.new(@disk_id))
+        expect(disk.datastore.name).to match(@datastore_pattern_maintenance_cluster)
+      ensure
+        delete_disk(cpi, @disk_id)
+    end
+
+    end
+    after do
+      turn_maintenance_off_for_all_hosts(cpi, @cluster_name_maintenance)
+    end
+  end
+
+  context 'when regex matches one or more of datastores that are accessible by all maintenace mode hosts in a cluster (datastore-*) ' do
+    let(:vm_type) do
+      {
+          'ram' => 512,
+          'disk' => 2048,
+          'cpu' => 1
+      }
+    end
+    let(:cpi) do
+      options = cpi_options(
+          'datacenters' => [
+              {
+                  'name' => @datacenter_name,
+                  'datastore_pattern' => @datastore_pattern_maintenance_cluster,
+                  'persistent_datastore_pattern' => @datastore_pattern_maintenance_cluster,
+                  'clusters' => [
+                      {
+                          @cluster_name_maintenance => {}
+                      },
+                  ]
+              }
+          ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    # We turn the maintenance mode ON for all of the hosts (rounding off to the floor) in a cluster
+    before do
+      turn_maintenance_on_for_all_hosts(cpi, @cluster_name_maintenance)
+    end
+    it 'cpi should fail to replicate stemcell (create vm and create ephemeral disk) on a datastore accessible by all maintenance mode hosts and which belongs to same cluster' do
```

Comment:
This it statement can be just 'cpi should fail to replicate stemcell (create vm and create ephemeral disk)' as rest of it is covered by context.
Similarly you can update it statements for other tests and remove duplication